gCNV/cnv_germline_case_filter_workflow.wdl

```diff
@@ -312,80 +312,73 @@ task ExtractPoNFreq {
-                                         df.end_panel_idx), "PANEL_COUNT"] = 0
-            panel_counts=[]
-            for i, r in df.loc[df_mask & (df.start_panel_idx !=
-                                         df.end_panel_idx)].iterrows():
-                overlapping_panel_events = panel_df_slice.iloc[int(r.start_panel_idx):int(r.end_panel_idx)]
-                overlapping_exon_lengths = (np.minimum(r.end_exon_idx,overlapping_panel_events.end_exon_idx) -
-                                           np.maximum(r.start_exon_idx,overlapping_panel_events.start_exon_idx))
-                overlapping_panel_fracs = overlapping_exon_lengths/(r.end_exon_idx - r.start_exon_idx)
-                overlapping_panel_events = overlapping_panel_events.loc[overlapping_panel_fracs>=~{overlap_thresh}]
-                panel_count =  len(set(overlapping_panel_events.sample_path))
-                panel_counts.append(panel_count)
-
-            df.loc[df_mask & (df.start_panel_idx !=
-                                         df.end_panel_idx), "PANEL_COUNT"] = panel_counts
-
-        n_panel_samples = len(set(df_panel.sample_path))
-        contigs = set(df.contig)
-        svtypes = set(df.svtype)
-        for contig in contigs:
-            for svtype in svtypes:
-                annotate_with_panel_count(df_panel, df, svtype, contig)
-
-        df.loc[:,"PANEL_FREQ"]=(df.PANEL_COUNT/n_panel_samples)
-
-        df_annotations = df[["contig","start","PANEL_FREQ","PANEL_COUNT"]].copy()
+        df_panel_expanded = get_exon_expanded_events(df_panel, intervals)
+
+        df_expanded_with_panel = df_expanded.join(df_panel_expanded, how="left", lsuffix="_sample", rsuffix="_panel")
+
+        df_expanded_with_panel['overlapping_panel_exon_start']=np.maximum(df_expanded_with_panel.event_exon_start_sample, df_expanded_with_panel.event_exon_start_panel)
+        df_expanded_with_panel['overlapping_panel_exon_end']=np.minimum(df_expanded_with_panel.event_exon_end_sample, df_expanded_with_panel.event_exon_end_panel)
+
+        df_expanded['event_exon_length']=df_expanded.event_exon_end - df_expanded.event_exon_start
+        df_expanded_with_panel['overlapping_panel_exon_length']=np.maximum(df_expanded_with_panel.overlapping_panel_exon_end - df_expanded_with_panel.overlapping_panel_exon_start,0).fillna(0)
+
+        df_panel_counts = df_expanded_with_panel.groupby(['ID_sample','sample_name_panel']).agg(
+            {'overlapping_panel_exon_length':'sum'}
+            ).reset_index('sample_name_panel').join(df_expanded.groupby('ID').agg({'event_exon_length':'sum'})
+        )
+
+        df_panel_counts['PANEL_COUNT'] = np.where(df_panel_counts.overlapping_panel_exon_length/df_panel_counts.event_exon_length>~{overlap_thresh}, 1, 0)
+        df_panel_counts = df_panel_counts.groupby(df_panel_counts.index).agg({'PANEL_COUNT':'sum'})
+
+        df = df.set_index('ID').join(df_panel_counts).fillna({'PANEL_COUNT':0}).astype({'PANEL_COUNT':int})
```

Comment:
You should be able to remove either this cast for `PANEL_COUNT` or the one in line 383